### PR TITLE
Allow to omit collection in bulk insertions

### PR DESCRIPTION
**Pull request by constantinius**
**Related Issue(s):**

- #110

**Description:**

Currently, each item in a bulk insertion must match the collection on which it is executed. This is different for single item insertions. Additionally, no check for the items ID and collection ID was performed, which is now rectified.

**PR Checklist:**

- [x] `pre-commit` hooks pass locally
- [ ] Tests pass (run `make test`)
- [ ] Documentation has been updated to reflect changes, if applicable, and docs build successfully (run `make docs`)
- [x] Changes are added to the [CHANGELOG](https://github.com/stac-utils/stac-fastapi/blob/main/CHANGES.md).

### CODE REVIEW

**Comment by constantinius** [comment]
Can't run tests locally, because I get this error in the docker build:

```
Step 8/8 : RUN python -m pip install -e .[dev,server] --user
 ---> Running in eec7acdacb2d
Obtaining file:///app
  Installing build dependencies: started
  Installing build dependencies: finished with status 'done'
  Checking if build backend supports build_editable: started
  Checking if build backend supports build_editable: finished with status 'done'
  Getting requirements to build editable: started
  Getting requirements to build editable: finished with status 'error'
  error: subprocess-exited-with-error
  
  × Getting requirements to build editable did not run successfully.
  │ exit code: 1
  ╰─> [3 lines of output]
      No `name` configuration, performing automatic discovery
      running egg_info
      error: Cannot update time stamp of directory 'stac_fastapi.pgstac.egg-info'
      [end of output]
  
  note: This error originates from a subprocess, and is likely not a problem with pip.
error: subprocess-exited-with-error
```

**Comment by jonhealy1** [comment]
Linting error in gh actions. I'm not sure why you are getting that error?

**Comment by vincentsarago** [comment]
I'm pretty sure the error is due to some permission error if you use Docker, I would make sure there is no `stac_fastapi.pgstac.egg-info` previously created somewhere

File: stac_fastapi/pgstac/transactions.py [hunk 1]
```diff
@@ -176,11 +178,17 @@ async def delete_collection(
...
 
 @attr.s
-class BulkTransactionsClient(AsyncBaseBulkTransactionsClient):
+class BulkTransactionsClient(AsyncBaseBulkTransactionsClient, ClientValidateMixIn):
     """Postgres bulk transactions."""
 
-    async def bulk_item_insert(self, items: Items, request: Request, **kwargs) -> str:
+    async def bulk_item_insert(
+        self, collection_id: str, items: Items, request: Request, **kwargs
+    ) -> str:
```
[the remark this answers is not in the document]
Comment (vincentsarago) [reply stac_fastapi/pgstac/transactions.py]: resolved by using `request.path_params["collection_id"]`

File: tests/clients/test_postgres.py [hunk 2]
```diff
@@ -407,6 +407,90 @@ async def test_create_bulk_items_already_exist_upsert(
...
+        del _item["collection"]
+        items[_item["id"]] = _item
+
+    payload = {"items": items, "method": "insert"}
+
+    resp = await app_client.post(
+        f"/collections/{coll['id']}/bulk_items",
+        json=payload,
+    )
+    assert resp.status_code == 400
```
Comment (vincentsarago) [inline tests/clients/test_postgres.py:430]: if we allow to omit the `collection_id` then the response should be 200

**Review by vincentsarago — APPROVED** [review]

**Comment by vincentsarago** [comment]
after talking with @ emmanuelmathot we agree that the collection_id is not needed for the items when submitting to the bulk items ingest endpoint 👍 

thanks for the pr @constantinius 🙏

**Pull request merged**